gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/SSOCookieFederationFilter.java

```diff
@@ -197,7 +204,7 @@ private void sendRedirectToLoginURL(HttpServletRequest request, HttpServletRespo
 
   @Override
   protected void handleValidationError(HttpServletRequest request, HttpServletResponse response, int status, String error) throws IOException {
-    if (error != null && error.startsWith("Token") && error.endsWith("disabled")) {
+    if (error != null && error.startsWith("Token") && (error.endsWith("disabled") || error.endsWith("exceeded idle timeout"))) {
```

Comment:
Maybe we could set a special request attribute to indicate the idle timeout, or extracting the string to constant would make this check more robust.

Reply:
Fixed. I declared new constants and used them wherever it was needed.

Reply:
@zeroflag - Does the new PS satisfies what you asked here?

Reply:
@smolnar82 yes, LGTM